src/compat/assumptions.h

```diff
@@ -11,13 +11,6 @@
 #include <cstddef>
 #include <limits>
 
-// Assumption: We assume a C++17 (ISO/IEC 14882:2017) compiler (minimum requirement).
-// Example(s): We assume the presence of C++17 features everywhere :-)
-// ISO Standard C++17 [cpp.predefined]p1:
-// "The name __cplusplus is defined to the value 201703L when compiling a C++
-//  translation unit."
-static_assert(__cplusplus >= 201703L, "C++17 standard assumed");
```

Comment:
I don't see the harm in leaving this? It could be that it's possible to compile certain parts of the code as c++14, but we want to document through code that c++17 is THE supported standard.

For ex, I have no idea, but it may currently be possible (or possible soon, after the next refactors) to build an app using libbitcoinkernel's headers in c++14 mode. If that's the case, we want to force that to be an error.

Reply:
To be clear, issues (sorta kinda) like this are the ones I'm trying to guard against here: https://github.com/bitcoin/bitcoin/pull/27930 .

In that case, the definition of things changed in c++20/23. Ideally we'd be able to select exactly one behavior we're relying on (old or new) and fail to compile otherwise.

But yeah, I get that the state of things isn't perfect, so I don't care too much.